lib/orbit/authentication/user.ex

```diff
@@ -1,25 +1,36 @@
 defmodule Orbit.Authentication.User do
+  alias Orbit.Authentication.UserPermission
   use Ecto.Schema
   import Ecto.Changeset
 
   @type t :: %__MODULE__{
           id: integer(),
           email: String.t(),
+          permissions: [UserPermission.t()],
           inserted_at: DateTime.t(),
           updated_at: DateTime.t()
         }
 
   schema "users" do
     field(:email, :string)
+    field(:permissions, {:array, UserPermission})
     timestamps()
   end
 
-  def create_changeset(struct, params \\ %{}) do
+  def changeset(struct, params \\ %{}) do
     struct
     |> cast(params, [
-      :email
+      :email,
+      :permissions
     ])
-    |> validate_required([:email])
+    |> validate_required([:email, :permissions])
+    |> validate_change(:permissions, fn :permissions, permissions ->
+      if permissions |> Enum.uniq() |> length() == length(permissions) do
```

Comment:
Since elixir lists are compared by element, can this instead be `if Enum.uniq(permissions) != permissions`?

Reply:
Hmm, so my one fear is that [the docs on `Enum.uniq/1`](https://hexdocs.pm/elixir/Enum.html#uniq/1) don't actually say that it preserves the order of elements. It probably does in practice, but I don't want to rely on undocumented behavior like that. In contrast, [`Enum.group_by/3`s documentation](https://hexdocs.pm/elixir/Enum.html#group_by/3) does specify that it preserves the relative order of elements in the resulting lists within each map value.